diff -ur stock/bzip2-1.0.8/Makefile bzip2-1.0.8/Makefile
--- stock/bzip2-1.0.8/Makefile	Sat Jul 13 10:50:05 2019
+++ bzip2-1.0.8/Makefile	Fri Nov 10 22:48:53 2023
@@ -75,31 +75,31 @@
 	if ( test ! -d $(PREFIX)/man ) ; then mkdir -p $(PREFIX)/man ; fi
 	if ( test ! -d $(PREFIX)/man/man1 ) ; then mkdir -p $(PREFIX)/man/man1 ; fi
 	if ( test ! -d $(PREFIX)/include ) ; then mkdir -p $(PREFIX)/include ; fi
-	cp -f bzip2 $(PREFIX)/bin/bzip2
-	cp -f bzip2 $(PREFIX)/bin/bunzip2
-	cp -f bzip2 $(PREFIX)/bin/bzcat
-	cp -f bzip2recover $(PREFIX)/bin/bzip2recover
+	cp bzip2 $(PREFIX)/bin/bzip2
+	cp bzip2 $(PREFIX)/bin/bunzip2
+	cp bzip2 $(PREFIX)/bin/bzcat
+	cp bzip2recover $(PREFIX)/bin/bzip2recover
 	chmod a+x $(PREFIX)/bin/bzip2
 	chmod a+x $(PREFIX)/bin/bunzip2
 	chmod a+x $(PREFIX)/bin/bzcat
 	chmod a+x $(PREFIX)/bin/bzip2recover
-	cp -f bzip2.1 $(PREFIX)/man/man1
+	cp bzip2.1 $(PREFIX)/man/man1
 	chmod a+r $(PREFIX)/man/man1/bzip2.1
-	cp -f bzlib.h $(PREFIX)/include
+	cp bzlib.h $(PREFIX)/include
 	chmod a+r $(PREFIX)/include/bzlib.h
-	cp -f libbz2.a $(PREFIX)/lib
+	cp libbz2.a $(PREFIX)/lib
 	chmod a+r $(PREFIX)/lib/libbz2.a
-	cp -f bzgrep $(PREFIX)/bin/bzgrep
+	cp bzgrep $(PREFIX)/bin/bzgrep
 	ln -s -f $(PREFIX)/bin/bzgrep $(PREFIX)/bin/bzegrep
 	ln -s -f $(PREFIX)/bin/bzgrep $(PREFIX)/bin/bzfgrep
 	chmod a+x $(PREFIX)/bin/bzgrep
-	cp -f bzmore $(PREFIX)/bin/bzmore
+	cp bzmore $(PREFIX)/bin/bzmore
 	ln -s -f $(PREFIX)/bin/bzmore $(PREFIX)/bin/bzless
 	chmod a+x $(PREFIX)/bin/bzmore
-	cp -f bzdiff $(PREFIX)/bin/bzdiff
+	cp bzdiff $(PREFIX)/bin/bzdiff
 	ln -s -f $(PREFIX)/bin/bzdiff $(PREFIX)/bin/bzcmp
 	chmod a+x $(PREFIX)/bin/bzdiff
-	cp -f bzgrep.1 bzmore.1 bzdiff.1 $(PREFIX)/man/man1
+	cp bzgrep.1 bzmore.1 bzdiff.1 $(PREFIX)/man/man1
 	chmod a+r $(PREFIX)/man/man1/bzgrep.1
 	chmod a+r $(PREFIX)/man/man1/bzmore.1
 	chmod a+r $(PREFIX)/man/man1/bzdiff.1
diff -ur stock/bzip2-1.0.8/Makefile-libbz2_so bzip2-1.0.8/Makefile-libbz2_so
--- stock/bzip2-1.0.8/Makefile-libbz2_so	Sat Jul 13 10:50:05 2019
+++ bzip2-1.0.8/Makefile-libbz2_so	Fri Nov 10 23:02:23 2023
@@ -36,7 +36,7 @@
 
 all: $(OBJS)
 	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.8 $(OBJS)
-	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.8
+	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c -Wl,-rpath,/usr/local/lib libbz2.so.1.0.8
 	rm -f libbz2.so.1.0
 	ln -s libbz2.so.1.0.8 libbz2.so.1.0
 
