diff -ur stock/bzip2-1.0.8/Makefile-libbz2_so bzip2-1.0.8/Makefile-libbz2_so
--- stock/bzip2-1.0.8/Makefile-libbz2_so	Sat Jul 13 10:50:05 2019
+++ bzip2-1.0.8/Makefile-libbz2_so	Fri Nov 10 23:02:23 2023
@@ -36,7 +36,7 @@
 
 all: $(OBJS)
 	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.8 $(OBJS)
-	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.8
+	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c -Wl,-rpath,/usr/local/lib libbz2.so.1.0.8
 	rm -f libbz2.so.1.0
 	ln -s libbz2.so.1.0.8 libbz2.so.1.0
 
