diff -ur -x configure -x aclocal.m4 -x ltmain.sh -x Makefile.in -x libtool.m4 -x ltoptions.m4 -x ltversion.m4 -x lt~obsolete.m4 -x config.h.in stock/Eterm-0.9.6/configure.ac Eterm-0.9.6/configure.ac
--- stock/Eterm-0.9.6/configure.ac	Tue Mar 15 21:16:59 2011
+++ Eterm-0.9.6/configure.ac	Sat Nov 11 00:40:17 2023
@@ -1037,10 +1037,10 @@
 AC_SUBST(GDB_CMD_FILE)
 
 CPPFLAGS=`eval eval eval eval eval echo "-I$includedir -I$prefix/include $CPPFLAGS"`
-CPPFLAGS=`echo $CPPFLAGS | tr ' ' '\n' | uniq | grep -v NONE | tr '\n' ' '`
+CPPFLAGS=`echo $CPPFLAGS | tr ' ' '\012' | uniq | grep -v NONE | tr '\012' ' '`
 CFLAGS=${CFLAGS--O}
 LDFLAGS=`eval eval eval eval eval echo "-L$libdir -L$prefix/lib ${LDFLAGS--O}"`
-LDFLAGS=`echo $LDFLAGS | tr ' ' '\n' | uniq | grep -v NONE | tr '\n' ' '`
+LDFLAGS=`echo $LDFLAGS | tr ' ' '\012' | uniq | grep -v NONE | tr '\012' ' '`
 LIBS="$GRLIBS $X_PRE_LIBS $LIBS $X_EXTRA_LIBS"
 
 AC_SUBST(CC)
diff -ur -x configure -x aclocal.m4 -x ltmain.sh -x Makefile.in -x libtool.m4 -x ltoptions.m4 -x ltversion.m4 -x lt~obsolete.m4 -x config.h.in stock/Eterm-0.9.6/src/Makefile.am Eterm-0.9.6/src/Makefile.am
--- stock/Eterm-0.9.6/src/Makefile.am	Tue Jan  6 01:05:36 2009
+++ Eterm-0.9.6/src/Makefile.am	Sat Nov 11 02:57:04 2023
@@ -36,7 +36,11 @@
 
 Eterm_SOURCES = main.c
 Eterm_DEPENDENCIES = libEterm.la
-Eterm_LDFLAGS = -rpath $(libdir):$(pkglibdir)
+IMLD = $(libdir)/imlib2/loaders
+Eterm_LDFLAGS = -rpath $(libdir):$(pkglibdir) -export-dynamic -dlopen \
+  $(IMLD)/argb.la $(IMLD)/bmp.la $(IMLD)/bz2.la $(IMLD)/jpeg.la \
+  $(IMLD)/lbm.la $(IMLD)/png.la $(IMLD)/pnm.la $(IMLD)/tga.la \
+  $(IMLD)/tiff.la $(IMLD)/xpm.la $(IMLD)/zlib.la
 Eterm_LDADD = libEterm.la 
 
 EXTRA_DIST = gdb.scr mmx_cmod.S sse2_cmod.c
diff -ur -x configure -x aclocal.m4 -x ltmain.sh -x Makefile.in -x libtool.m4 -x ltoptions.m4 -x ltversion.m4 -x lt~obsolete.m4 -x config.h.in stock/Eterm-0.9.6/src/defaultfont.h Eterm-0.9.6/src/defaultfont.h
--- stock/Eterm-0.9.6/src/defaultfont.h	Tue Jan  6 01:05:36 2009
+++ Eterm-0.9.6/src/defaultfont.h	Sat Nov 11 00:40:17 2023
@@ -8,7 +8,10 @@
 # endif
 
 #ifdef HAVE_NL_LANGINFO
+#undef MALLOC
 # include <langinfo.h>
+#undef MALLOC
+#define MALLOC(sz) malloc(sz)
 #endif
 
 
diff -ur -x configure -x aclocal.m4 -x ltmain.sh -x Makefile.in -x libtool.m4 -x ltoptions.m4 -x ltversion.m4 -x lt~obsolete.m4 -x config.h.in stock/Eterm-0.9.6/src/events.c Eterm-0.9.6/src/events.c
--- stock/Eterm-0.9.6/src/events.c	Tue Mar 15 21:16:59 2011
+++ Eterm-0.9.6/src/events.c	Sat Nov 11 00:40:17 2023
@@ -217,7 +217,9 @@
     }
     if (BITFIELD_IS_SET(vt_options, VT_OPTIONS_URG_ALERT)) {
         wm_hints = XGetWMHints(Xdisplay, TermWin.parent);
+#ifdef XUrgencyHint
         wm_hints->flags &= ~XUrgencyHint;
+#endif
         XSetWMHints(Xdisplay, TermWin.parent, wm_hints);
         XFree(wm_hints);
     }
@@ -498,7 +500,9 @@
 #endif
         if (BITFIELD_IS_SET(vt_options, VT_OPTIONS_URG_ALERT)) {
             wm_hints = XGetWMHints(Xdisplay, TermWin.parent);
+#ifdef XUrgencyHint
             wm_hints->flags &= ~XUrgencyHint;
+#endif
             XSetWMHints(Xdisplay, TermWin.parent, wm_hints);
             XFree(wm_hints);
         }
diff -ur -x configure -x aclocal.m4 -x ltmain.sh -x Makefile.in -x libtool.m4 -x ltoptions.m4 -x ltversion.m4 -x lt~obsolete.m4 -x config.h.in stock/Eterm-0.9.6/src/libscream.c Eterm-0.9.6/src/libscream.c
--- stock/Eterm-0.9.6/src/libscream.c	Fri Nov 20 19:58:10 2009
+++ Eterm-0.9.6/src/libscream.c	Sat Nov 11 00:40:17 2023
@@ -45,6 +45,10 @@
 #include "config.h"
 #include "feature.h"
 
+#if defined(__svr4__) && defined(__i386__)
+typedef int ssize_t;
+#endif
+
 /* use libast if we have it */
 #ifdef DEBUG_ESCREEN
 #  include <libast.h>
diff -ur -x configure -x aclocal.m4 -x ltmain.sh -x Makefile.in -x libtool.m4 -x ltoptions.m4 -x ltversion.m4 -x lt~obsolete.m4 -x config.h.in stock/Eterm-0.9.6/src/screen.c Eterm-0.9.6/src/screen.c
--- stock/Eterm-0.9.6/src/screen.c	Tue Mar 15 21:16:59 2011
+++ Eterm-0.9.6/src/screen.c	Sat Nov 11 00:40:17 2023
@@ -1553,7 +1553,9 @@
 
     if (BITFIELD_IS_SET(vt_options, VT_OPTIONS_URG_ALERT)) {
         wm_hints = XGetWMHints(Xdisplay, TermWin.parent);
+#ifdef XUrgencyHint
         wm_hints->flags |= XUrgencyHint;
+#endif
         XSetWMHints(Xdisplay, TermWin.parent, wm_hints);
         XFree(wm_hints);
     }
diff -ur -x configure -x aclocal.m4 -x ltmain.sh -x Makefile.in -x libtool.m4 -x ltoptions.m4 -x ltversion.m4 -x lt~obsolete.m4 -x config.h.in stock/Eterm-0.9.6/src/term.c Eterm-0.9.6/src/term.c
--- stock/Eterm-0.9.6/src/term.c	Tue Mar 15 21:16:59 2011
+++ Eterm-0.9.6/src/term.c	Sat Nov 11 00:40:17 2023
@@ -723,7 +723,9 @@
 
                         /* Tab key is normal unless it's shift-tab. */
                     case XK_Tab:
+#ifdef XK_ISO_Left_Tab
                     case XK_ISO_Left_Tab:
+#endif
                         if (shft) {
                             len = 3;
                             strcpy(kbuf, "\033[Z");
diff -ur -x configure -x aclocal.m4 -x ltmain.sh -x Makefile.in -x libtool.m4 -x ltoptions.m4 -x ltversion.m4 -x lt~obsolete.m4 -x config.h.in stock/Eterm-0.9.6/src/windows.c Eterm-0.9.6/src/windows.c
--- stock/Eterm-0.9.6/src/windows.c	Tue Mar 15 21:16:59 2011
+++ Eterm-0.9.6/src/windows.c	Sat Nov 11 00:40:17 2023
@@ -504,7 +504,12 @@
 
         atom = XInternAtom(Xdisplay, "UTF8_STRING", False);
         XChangeProperty(Xdisplay, TermWin.parent, props[PROP_EWMH_STARTUP_ID], atom, 8, PropModeReplace, tmp, strlen(tmp) + 1);
+
+#ifdef __svr4__
+        putenv("DESKTOP_STARTUP_ID="); 
+#else
         unsetenv("DESKTOP_STARTUP_ID");
+#endif
     }
 
     /* Set window opacity if needed. */
