diff -ur stock/freetype-2.3.9/builds/unix/ftsystem.c freetype-2.3.9/builds/unix/ftsystem.c
--- stock/freetype-2.3.9/builds/unix/ftsystem.c	Wed Jan 21 21:51:10 2009
+++ freetype-2.3.9/builds/unix/ftsystem.c	Thu Sep 28 05:18:19 2023
@@ -25,6 +25,8 @@
 #include FT_TYPES_H
 #include FT_INTERNAL_STREAM_H
 
+typedef int ssize_t;
+
   /* memory-mapping includes and definitions */
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
