diff -ur stock/gcc-3.4.6/Makefile.in gcc-3.4.6/Makefile.in
--- stock/gcc-3.4.6/Makefile.in	Thu Sep 23 17:43:53 2004
+++ gcc-3.4.6/Makefile.in	Fri Nov 10 06:22:02 2023
@@ -23655,7 +23655,7 @@
 	@r=`${PWD_COMMAND}`; export r; \
 	echo "Checking multilib configuration..."; \
 	$(CC_FOR_TARGET) --print-multi-lib > multilib.tmp 2> /dev/null ; \
-	$(SHELL) $(srcdir)/move-if-change multilib.tmp multilib.out ; \
+	$(SHELL) $(srcdir)/move-if-change multilib.tmp multilib.out ; 
 
 # Rebuilding Makefile.in, using autogen.
 AUTOGEN = autogen
diff -ur stock/gcc-3.4.6/gcc/configure gcc-3.4.6/gcc/configure
--- stock/gcc-3.4.6/gcc/configure	Wed Jun  1 09:32:31 2005
+++ gcc-3.4.6/gcc/configure	Fri Nov 10 03:32:02 2023
@@ -2757,6 +2757,7 @@
   eval ac_cv_prog_cc_${ac_cc}_c_o=yes
   if test "x$CC" != xcc; then
     # Test first that cc exists at all.
+    if false; then
     if { ac_try='cc -c conftest.$ac_ext >&5'
   { (eval echo "$as_me:$LINENO: \"$ac_try\"") >&5
   (eval $ac_try) 2>&5
@@ -2781,6 +2782,7 @@
 	# cc exists but doesn't like -o.
 	eval ac_cv_prog_cc_${ac_cc}_c_o=no
       fi
+    fi
     fi
   fi
 else
diff -ur stock/gcc-3.4.6/libstdc++-v3/testsuite/testsuite_hooks.cc gcc-3.4.6/libstdc++-v3/testsuite/testsuite_hooks.cc
--- stock/gcc-3.4.6/libstdc++-v3/testsuite/testsuite_hooks.cc	Thu Dec 23 13:53:36 2004
+++ gcc-3.4.6/libstdc++-v3/testsuite/testsuite_hooks.cc	Fri Nov 10 06:26:14 2023
@@ -34,6 +34,10 @@
 #include <unistd.h>
 #include <sys/time.h>
 #include <sys/resource.h>
+#ifdef __svr4__
+int getrlimit(int resource, struct rlimit *rlp);
+int setrlimit(int resource, const struct rlimit *rlp);
+#endif
 #endif
 #include <list>
 #include <string>
