diff -ur -x configure -x aclocal.m4 -x ltmain.sh -x Makefile.in -x libtool.m4 -x ltoptions.m4 -x ltversion.m4 -x lt~obsolete.m4 -x config.h.in -x sysdefs.h stock/libast-0.7/configure.in libast-0.7/configure.in
--- stock/libast-0.7/configure.in	Mon Jan 16 20:23:49 2006
+++ libast-0.7/configure.in	Thu Nov 16 04:25:01 2023
@@ -12,6 +12,8 @@
 AC_SUBST(AUTOHEADER)
 AC_SUBST(AUTOMAKE)
 
+AC_CONFIG_MACRO_DIR([.])
+
 SAVE_LIBTOOLIZE="$LIBTOOLIZE"
 SAVE_ACLOCAL="$ACLOCAL"
 SAVE_AUTOCONF="$AUTOCONF"
diff -ur -x configure -x aclocal.m4 -x ltmain.sh -x Makefile.in -x libtool.m4 -x ltoptions.m4 -x ltversion.m4 -x lt~obsolete.m4 -x config.h.in -x sysdefs.h stock/libast-0.7/include/libast/types.h libast-0.7/include/libast/types.h
--- stock/libast-0.7/include/libast/types.h	Mon Jan 23 10:25:28 2006
+++ libast-0.7/include/libast/types.h	Thu Nov 16 04:19:33 2023
@@ -779,6 +779,10 @@
  */
 typedef spif_uint16_t spif_sockport_t;
 
+#ifdef __svr4__
+typedef int socklen_t;
+#endif
+
 /**
  * The length of a socket address structure.
  *
diff -ur -x configure -x aclocal.m4 -x ltmain.sh -x Makefile.in -x libtool.m4 -x ltoptions.m4 -x ltversion.m4 -x lt~obsolete.m4 -x config.h.in -x sysdefs.h stock/libast-0.7/include/libast/types.h.in libast-0.7/include/libast/types.h.in
--- stock/libast-0.7/include/libast/types.h.in	Tue Dec 28 22:39:10 2004
+++ libast-0.7/include/libast/types.h.in	Thu Nov 16 04:10:03 2023
@@ -779,6 +779,10 @@
  */
 typedef spif_uint16_t spif_sockport_t;
 
+#ifdef __svr4__
+typedef int socklen_t;
+#endif
+
 /**
  * The length of a socket address structure.
  *
diff -ur -x configure -x aclocal.m4 -x ltmain.sh -x Makefile.in -x libtool.m4 -x ltoptions.m4 -x ltversion.m4 -x lt~obsolete.m4 -x config.h.in -x sysdefs.h stock/libast-0.7/include/libast.h libast-0.7/include/libast.h
--- stock/libast-0.7/include/libast.h	Mon Jan 23 10:17:26 2006
+++ libast-0.7/include/libast.h	Thu Nov 16 04:10:03 2023
@@ -107,6 +107,10 @@
 #  endif
 #endif
 
+#ifdef __svr4__
+#include <varargs.h>
+#endif
+
 #include <libast/types.h>
 #include <libast/obj.h>
 
diff -ur -x configure -x aclocal.m4 -x ltmain.sh -x Makefile.in -x libtool.m4 -x ltoptions.m4 -x ltversion.m4 -x lt~obsolete.m4 -x config.h.in -x sysdefs.h stock/libast-0.7/src/snprintf.c libast-0.7/src/snprintf.c
--- stock/libast-0.7/src/snprintf.c	Fri Jul 23 15:34:05 2004
+++ libast-0.7/src/snprintf.c	Thu Nov 16 04:10:03 2023
@@ -9,6 +9,10 @@
 # undef HAVE_STDARG_H
 #endif
 
+#ifdef __svr4__
+#include <sys/types.h>
+#endif
+
 static const char __attribute__((unused)) cvs_ident[] = "$Id: snprintf.c,v 1.6 2004/07/23 21:38:39 mej Exp $";
 
 /* 
diff -ur -x configure -x aclocal.m4 -x ltmain.sh -x Makefile.in -x libtool.m4 -x ltoptions.m4 -x ltversion.m4 -x lt~obsolete.m4 -x config.h.in -x sysdefs.h stock/libast-0.7/src/socket.c libast-0.7/src/socket.c
--- stock/libast-0.7/src/socket.c	Fri Jul 23 15:34:05 2004
+++ libast-0.7/src/socket.c	Thu Nov 16 04:10:03 2023
@@ -721,7 +721,11 @@
         hinfo = gethostbyname(SPIF_CHARPTR_C(SPIF_STR_STR(hostname)));
     } while ((tries <= 3) && (hinfo == NULL) && (h_errno == TRY_AGAIN));
     if (hinfo == NULL) {
+	#ifndef __svr4__
         libast_print_error("Unable to resolve hostname \"%s\" -- %s\n", SPIF_STR_STR(hostname), hstrerror(h_errno));
+	#else
+        libast_print_error("Unable to resolve hostname \"%s\" -- h errno %d\n", SPIF_STR_STR(hostname), h_errno);
+	#endif
         return SPIF_NULL_TYPE(ipsockaddr);
     }
 
