diff -ur -x Makefile.in -x aclocal.m4 -x configure -x ltmain.sh -x libtool -x libtool.m4 -x ltversion.m4 stock/libpng-1.6.34/Makefile.am libpng-1.6.34/Makefile.am
--- stock/libpng-1.6.34/Makefile.am	Fri Aug  4 10:32:58 2017
+++ libpng-1.6.34/Makefile.am	Fri Nov 10 22:31:14 2023
@@ -115,7 +115,7 @@
 
 nodist_libpng@PNGLIB_MAJOR@@PNGLIB_MINOR@_la_SOURCES = pnglibconf.h
 
-libpng@PNGLIB_MAJOR@@PNGLIB_MINOR@_la_LDFLAGS = -no-undefined -export-dynamic \
+libpng@PNGLIB_MAJOR@@PNGLIB_MINOR@_la_LDFLAGS = -export-dynamic \
 	-version-number @PNGLIB_MAJOR@@PNGLIB_MINOR@:@PNGLIB_RELEASE@:0
 
 if HAVE_LD_VERSION_SCRIPT
